rclpy/src/rclpy/_rclpy.c

```diff
@@ -866,6 +866,65 @@ rclpy_get_node_logger_name(PyObject * Py_UNUSED(self), PyObject * args)
   return PyUnicode_FromString(node_logger_name);
 }
 
+/// Get the name of the logger associated with the node of the publisher.
+/**
+ * Raises ValueError if pypublisher is not a publisher capsule
+ *
+ * \param[in] pypublisher Capsule pointing to the publisher to get the logger name of
+ * \return logger_name, or
+ * \return None on failure
+ */
+static PyObject *
+rclpy_get_publisher_logger_name(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pypublisher;
+  if (!PyArg_ParseTuple(args, "O", &pypublisher)) {
+    return NULL;
+  }
+
+  rclpy_publisher_t * pub = rclpy_handle_get_pointer_from_capsule(pypublisher, "rclpy_publisher_t");
+  if (NULL == pub) {
+    return NULL;
+  }
+
+  const char * node_logger_name = rcl_node_get_logger_name(pub->node);
+  if (NULL == node_logger_name) {
+    Py_RETURN_NONE;
+  }
```

Comment:
This is fine, though I think that a bit unnecessary.

IMO, a nicer approach would have been the following:

```python3
callbacks = PublisherEventCallbacks()  # default callbacks is true, but they're not yet defined
# Default callbacks are added here, if `event_callbacks._use_default_callbacks` is true, and the user doesn't explicitly define that callback.
publisher = self.node.create_publisher(EmptyMsg, 'test_topic', 10, event_callbacks=callbacks)
```

In both cases the result the user sees is the same, so I won't block on the change.

Reply:
I thought about doing it this way, but with `event_callbacks.incompatible_qos` no longer `None`, it's harder to determine later on if it's the default one or a user provided one, which is needed to handle the `UnsupportedEventTypeError` exception differently.

Reply:
> I thought about doing the way you mentioned, but with event_callbacks.incompatible_qos no longer None, it's harder to determine later on if it's the default one or a user provided one, which is needed to handle the UnsupportedEventTypeError exception differently.

It could initially be `None`, except the user provided explicitly one.
If it's set to `None` and use `default_callbacks` is true, the event is registered with the default callback when the publisher is created.

I think the approach I'm mentioning is exactly the one used in `rclcpp`.